File: Misc/kytdkut_Import all media in project directory - query padding.py
# description Import all media in project directory - query padding seconds
# @version 1.0
# @author kytdkut

# Sound in Words - only intended for internal use
# this is useful when creating RPP files via scripting
# this script imports all the wav media contained in the media_path folder and separates it using the prompted value (kind of)

import reaper_python
from pathlib import Path
from sys import exit

media_files = []
user_input = RPR_GetUserInputs("Insert padding amount", 1, "Amount in seconds:", 2, 100)
right_padding = float(user_input[4])

(path, len) = RPR_GetProjectPath("", 512)
media_path = Path(path)

for file in media_path.glob("[!$._]*.wav"):
    media_files.append(file)

media_files = sorted(media_files)


def main():
    RPR_Undo_BeginBlock()
    
    for file in media_files:
        RPR_InsertMedia(file, 0)
    
    selected_track = RPR_GetSelectedTrack(0, 0)
    num_items = RPR_CountTrackMediaItems(selected_track)
    item_durs = []
    
    
    for i in range(num_items): # get item lengths
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        item_durs.append(RPR_GetMediaItemInfo_Value(current_item, 'D_LENGTH'))
    for i in reversed(range(num_items)): # reposition items leaving 2 times its length in silence afterwards
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        current_item_pos = RPR_GetMediaItemInfo_Value(current_item, 'D_POSITION')
        RPR_SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos * 3)
    for i in reversed(range(num_items)): # add right_padding seconds between items
        current_item = RPR_GetTrackMediaItem(selected_track, i)
        current_item_pos = RPR_GetMediaItemInfo_Value(current_item, 'D_POSITION')
        RPR_SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos + right_padding * i)

    RPR_Undo_EndBlock("Import all media in project directory", -1)


if user_input[0] == 1:
    RPR_SetEditCurPos(0, 1, 1)
    main()
    RPR_SetEditCurPos(0, 1, 1)

File: Misc/kytdkut_Import all media in project directory - query padding.lua
-- description Import all media in project directory - query padding in seconds
-- @version 1.02
-- @author kytdkut

-- Sound in Words - only intended for internal use
local project_path = reaper.GetProjectPath(0)
local files = {}
local retval, padding_amount = reaper.GetUserInputs("Insert padding amount", 1, "Amount in seconds:", 2, 100)
local selected_track = reaper.GetSelectedTrack(0, 0)

function GetFilesInMediaFolder()
    local i = 0
    repeat
        local file = reaper.EnumerateFiles(project_path, i)
        if file then
            local ext = file:match('.-%.(.*)')
            if reaper.IsMediaExtension(ext, false) then
                files[i + 1] = file -- arrays in lua start on 1
            end
            i = i + 1
        end
    until not file
    table.sort(files)
end

function RepositionItems()
    local num_items = reaper.CountTrackMediaItems(selected_track)
    -- first repositioning - 2x item dur
    for i = num_items - 1, 1, -1 do -- reverse iterate
        local current_item = reaper.GetTrackMediaItem(selected_track, i)
        local current_item_pos = reaper.GetMediaItemInfo_Value(current_item, 'D_POSITION')
        reaper.SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos * 3)
    end
    -- second repositioning - add padding_amount
    for i = num_items - 1, 1, -1 do
        local current_item = reaper.GetTrackMediaItem(selected_track, i)
        local current_item_pos = reaper.GetMediaItemInfo_Value(current_item, 'D_POSITION')
        reaper.SetMediaItemInfo_Value(current_item, 'D_POSITION', current_item_pos + padding_amount * i)
    end
end

if retval and selected_track then
    reaper.Undo_BeginBlock()
    reaper.PreventUIRefresh(1)
    reaper.SetEditCurPos(0, 1, 1)
    GetFilesInMediaFolder()
    for i in pairs(files) do
        reaper.InsertMedia(project_path .. "/"  .. files[i], 0)
    end
    RepositionItems()
    reaper.SetEditCurPos(0, 1, 1)
    reaper.PreventUIRefresh(-1)
    reaper.Undo_EndBlock("Import all media in project directory - query padding", -1)
end
